### PR TITLE
Update docs regarding pullSecrets

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Update docs regarding pullSecrets

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/README.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 - [Deployment Guide](#deployment-guide)
   - [Solution overview](#solution-overview)
   - [Minimum Requirements](#minimum-requirements)
   - [Support Matrix](#support-matrix)
   - [Installation with Helm](./docs/Installation_with_Helm.md)
-  - [Non Helm Installation](./docs/Non_Helm_Installation.md) 
+  - [Non Helm Installation](./docs/Non_Helm_Installation.md)
   - [Adding Additional FluentD Plugins](./docs/Additional_Fluentd_Plugins.md)
   - [Advanced Configuration/Best Practices](./docs/Best_Practices.md)
+  - [Authenticating with docker registry](./docs/Authenticating_with_docker_registry.md)
```
Comment (perk-sumo) [inline deploy/README.md:16]: I don't see this file in the PR 🤔

Comment (perk-sumo) [inline deploy/README.md:16]: I'd call it either `Docker Hub` or `container registry` with the preference for the latter.

File: deploy/docs/Authenticating_with_docker_registry.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Authenticating with docker registry
+
+Sumo Logic docker images used in the collection are currently hosted on hub.docker.com which
```
Comment (perk-sumo) [inline deploy/docs/Authenticating_with_docker_registry.md:3]:
```suggestion
Sumo Logic container images used in the collection are currently hosted on hub.docker.com which
```

File: deploy/docs/Authenticating_with_docker_registry.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+[requires authentication in order to provide higher quota for image pulls][docker-rate-limit].
+
+In order to authenticate with docker registry hosted at hub.docker.com when using
+helm installation one can use `sumologic.pullSecrets`, to pass kubernetes secret
```
Comment (perk-sumo) [inline deploy/docs/Authenticating_with_docker_registry.md:7]:
```suggestion
helm installation one can use `sumologic.pullSecrets`, to pass Kubernetes secret
```

File: deploy/docs/Authenticating_with_docker_registry.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Authenticating with docker registry
+
+Sumo Logic docker images used in the collection are currently hosted on hub.docker.com which
+[requires authentication in order to provide higher quota for image pulls][docker-rate-limit].
+
+In order to authenticate with docker registry hosted at hub.docker.com when using
```
Comment (perk-sumo) [inline deploy/docs/Authenticating_with_docker_registry.md:6]:
```suggestion
In order to authenticate with container registry hosted at hub.docker.com when using
```

File: deploy/README.md (hunk 1 above)
Comment (perk-sumo) [reply deploy/README.md:16]:
```suggestion
  - [Authenticating with container registry](./docs/Authenticating_with_docker_registry.md)
```

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -932,6 +943,13 @@ prometheus-operator:
   ## Ensure we use pre 1.14 recording rules consistently as current content depends on them.
   kubeTargetVersionOverride: 1.13.0-0
 
+  # global:
+    ## Reference to one or more secrets to be used when pulling images
+    ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+    ##
+    # imagePullSecrets:
+    # - name: "image-pull-secret"
```
Comment (perk-sumo) [inline deploy/helm/sumologic/values.yaml:951]:
```suggestion
    #   - name: "image-pull-secret"
```

File: deploy/docs/Authenticating_with_docker_registry.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Authenticating with container registry
+
+Sumo Logic container images used in the collection are currently hosted on hub.docker.com which
```
Comment (sumo-drosiek) [inline deploy/docs/Authenticating_with_docker_registry.md:3]:
```suggestion
Sumo Logic container images used in the collection are currently hosted on [Docker Hub](hub.docker.com) which
```

File: deploy/docs/Authenticating_with_docker_registry.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+[requires authentication in order to provide higher quota for image pulls][docker-rate-limit].
+
+In order to authenticate with container registry hosted at hub.docker.com when using
```
Comment (sumo-drosiek) [inline deploy/docs/Authenticating_with_docker_registry.md:6]:
```suggestion
In order to authenticate with container registry hosted at [Docker Hub](https://hub.docker.com/) when using
```

**Review by sumo-drosiek — COMMENTED** [review]
LGTM, but wait for @drduke1 review

**Review by drduke1 — APPROVED** [review]

**Pull request merged**